src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/UIPropertyDataProducer.cs

```diff
@@ -58,6 +58,11 @@ public static IEntityValue CreateUIPropertyValue(IQueryExecutionContext executio
                 newUIProperty.ConfigurationIndependent = !property.IsConfigurationDependent();
             }
 
+            if (requestedProperties.IsReadOnly)
+            {
+                newUIProperty.IsReadOnly = property.ReadOnly;
+            }
```

Comment:
The changes look good. I'm just commenting with a more... personal question. For this method, do you think it is easier to read as a series of if-statements (current format) or if there was a data map between the properties of `BaseProperty` and `UIPropertyValue` (think constructor or extension method or something like dependency injection)? I'm just asking out of personal preference. If-statements leave me with the, "If it isn't set (statement is false), then what is the value?"

Reply:
> If it isn't set (statement is false), then what is the value?

The semantics here are that any non-requested property should not be set, and by doing so any attempt to actually use the property will trigger a runtime exception. So there's no need to worry about an un-set value.

Reply:
Interesting! That is a convention that is not conveyed well in this language. :-( Are the properties something similar to [`Lazy<T>`](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization)? The benefit of those is thread safety and such. And, you could set them up so that if [IsValueCreated](https://docs.microsoft.com/en-us/dotnet/api/system.lazy-1.isvaluecreated?view=net-5.0) is false, then throw the runtime exception.

Reply:
It's a convention for this library. It asks a little of the author, but gives better performance than something like lazy.

It's similar to EF from what I remember of using it many years ago. If you attempt to use data you didn't request, it can error at runtime.

Ideally this would be a compile time check, but I can't see how that's possible with the C# type system.